pyproject.toml

```diff
@@ -1,11 +1,64 @@
 [build-system]
-# Minimum requirements for the build system to execute.
+# Uncomment this line in order to build with Meson by default
+#build-backend = "mesonpy"
 requires = [
+    # setuptools, wheel and Cython are needed for the setup.py based build
     "setuptools==59.2.0",
+    # `wheel` is needed for non-isolated builds, given that `meson-python`
+    # doesn't list it as a runtime requirement (at least in 0.11.0) - it's
+    # likely to be removed as a dependency in meson-python 0.12.0.
     "wheel==0.37.0",
     "Cython>=0.29.30,<3.0",
+    "meson-python>=0.10.0",
 ]
 
+[project]
+name = "numpy"
+
+# Using https://peps.python.org/pep-0639/
+# which is still in draft
+license = {text = "BSD-3-Clause"}
+license-files.paths = [
+    "LICENSE.txt",
+    "LICENSES_bundles.txt"
+]
```

Comment:
Unsupported versions of setuptools that are NOT version 59 as pinned in this file, add pyproject.toml support to begin with -- and don't like this PEP 639 draft metadata key because reasons.

Reply:
Basically, apparently it's setuptools' job (even if setuptools is supposed to use setup.py) to refuse to build anything at all, if a new informational metadata value is added to the standard and setuptools doesn't support it.

I guess that's sort of defensible. The spec says:
> When specifying project metadata, tools MUST adhere and honour the metadata as specified in this PEP. If metadata is improperly specified then tools MUST raise an error to notify the user about their mistake.

@abravalheri was presumably relying on that wording when implementing setuptools' hard error mode there?

Reply:
Hi @eli-schwartz , if setuptools is invoked in a project that contains `pyproject.toml` it will try to read metadata/configuration from it.

It also will try to follow the spec as close as possible.

Reply:
Thanks for the explanations. I'll open a PR to comment out the problematic items. 

Side note: it'd be great if PEP 639 was hurried along, it's quite annoying that PyPI doesn't properly support PEP 621 license info (mainly because it's not sufficient, bad spec), and is waiting for PEP 639 to get something sensible implemented. Right now are choices are basically to ignore any spec and do whatever `setuptools` does, or to get the mess that you see in the sidebar of https://pypi.org/project/scipy/. Context in https://github.com/mesonbuild/meson-python/issues/129.

Reply:
Done in gh-22687